data_diff/diff_tables.py

```diff
@@ -227,6 +244,24 @@ def diff_tables(self, table1: TableSegment, table2: TableSegment) -> DiffResult:
         if self.bisection_factor < 2:
             raise ValueError("Must have at least two segments per iteration (i.e. bisection_factor >= 2)")
 
+        table1, table2 = self._threaded_call("with_schema", [table1, table2])
+        for c in table1._relevant_columns:
+            if c not in table1._schema:
+                raise ValueError(f"Column '{c}' not found in schema for table {table1}")
+            if c not in table2._schema:
+                raise ValueError(f"Column '{c}' not found in schema for table {table2}")
+
+            # Update schemas to minimal mutual precision
+            col1 = table1._schema[c]
+            col2 = table2._schema[c]
+            if isinstance(col1, PrecisionType):
+                if not isinstance(col2, PrecisionType):
+                    raise TypeError(f"Incompatible types for column {c}:  {col1} <-> {col2}")
+
+                min_precision = min(col1.precision, col2.precision)
+                table1._schema[c] = col1.replace(precision=min_precision)
+                table2._schema[c] = col2.replace(precision=min_precision)
```

Comment:
😍 😍 😍 😍 😍 😍 

If `min_precision` isn't equal to both precisions, I think you should output an information warning here!

Reply:
Good point!